auth/password_authenticator.cc

```diff
@@ -66,6 +66,7 @@ constexpr std::string_view password_authenticator_name("org.apache.cassandra.aut
 
 // name of the hash column.
 static constexpr std::string_view SALTED_HASH = "salted_hash";
+static constexpr std::string_view OPTIONS = "options";
```

Comment:
I don't understand why this is related to the authenticator. Authenticator options should be about how the authenticator works, and we shouldn't need per-authenticator code for something generic like session configuration.

There are two options (hah!) here:
1. CREATE ROLE WITH OPTIONS is not used by any authenticator, and we find something to convince us that OPTIONS is not related to the authenticator. In this case we should redirect OPTIONS from the authenticator to the role manager.
2. CREATE ROLE WITH OPTIONS _is_ used by some authenticator or other, and so we create or reuse a keyword for out map. For example CREATE ROLE WITH SESSION OPTIONS = { ... }.

@dekimir any wisdom about roles here? Can we store stuff in the roles table even if the user uses a non-default role manager?

Reply:
Ok, I got more familiar with role manager, and I pinpointed the source of confusion. The design of our code really heavily suggests that it's the authenticator that stores custom **per-role** options. Here's a snippet from `list_roles_statement::execute`:
```cpp
    return do_for_each(sorted_roles, [&results, &rm, &a](const sstring& role) {
        return when_all_succeed(
                rm.can_login(role),
                rm.is_superuser(role),
                a.query_custom_options(role)).then_unpack([&results, &role](
                ...
```
For each role, we extract some information from the role manager, but then query the authenticator (`a`) for custom per-role options.

So, if I follow the review correctly, I should instead extend the role manager to use the already existing `options` table of `system_auth.roles` to store per-role options, and then the password authenticator could get options support for free, since the table it uses would already be populated.

I think that the main source of my confusion is that `system_auth.roles` is used both for general management of roles, regardless of which authenticator is used, and for storing PasswordAuthenticator-specific data, like passwords.

Also, the current status of authenticators supporting their custom options is that none of them do. And from what I saw in enterprise sasl authenticator, it claims supporting additional options, but there are no actual options implemented, so it basically just supports options by always returning an empty option set if asked.

@avikivity I now know what caused my confusion, but I still don't really have a clear idea on how to proceed. Should I rework `ALTER ROLE x WITH options = ` to fill our table instead of going straight to authenticators, which do not implement custom options? The current infrastructure is quite heavily based on the fact that these options belong to authenticators (e.g. the underlying authenticators all validate if they support options, and refuse to alter roles if they don't)